src/cluster_legacy.c

```diff
@@ -2937,6 +3069,11 @@ int clusterProcessPacket(clusterLink *link) {
         sender->flags |= CLUSTER_NODE_EXTENSIONS_SUPPORTED;
     }
 
+    /* Checks if the node supports light message hdr */
+    if (sender && (flags & CLUSTER_NODE_LIGHT_HDR_SUPPORTED)) {
+        sender->flags |= CLUSTER_NODE_LIGHT_HDR_SUPPORTED;
+    }
```

Comment:
Is it possible that a node is downgraded (due to some problem after upgrade) and clears this flag? We should probably also clear this flag in that case.

@PingXie @madolson Why don't we just take the sender's node flags as they are here, i.e. `sender->flags = flags`? Any historical reason?

Reply:
>Is it possible that a node is downgraded (due to some problem after upgrade) and clears this flag? We should probably also clear this flag in that case.
 Yes we can unset it if it is not in the flag like
 ```
    if (sender) {
        if (flags & CLUSTER_NODE_LIGHT_HDR_SUPPORTED) {
            sender->flags |= CLUSTER_NODE_LIGHT_HDR_SUPPORTED;
        } else {
            sender->flags &= ~CLUSTER_NODE_LIGHT_HDR_SUPPORTED;
        }
    }
   ```
   
   We might want to do the same with the ping extension flags in this case?

Reply:
Good.



> We might want to do the same with the ping extension flags in this case?

Yes, but not in this PR. We can open an issue for it.

We had some bugs even for the upgrading scenario when ping extensions were added in 7.0(?). I want to make sure we're safer this time.

Reply:
>Is it possible that a node is downgraded (due to some problem after upgrade) and clears this flag? We should probably also clear this flag in that case.

Agreed.

>Why don't we just take the sender's node flags as they are here, i.e. sender->flags = flags? Any historical reason?

I am guessing there was no such need in the past. It is until the PING extension and now this newer message format that we have a forward compat issue to worry about

Reply:
AFAIK we make no guarantees about forward compatibility, so as Ping messaged there was no need for backwards compatibility. I guess also it would be safe to spin up new nodes on an older version, but you can't downgrade existing nodes and re-use the nodes.conf.

Reply:
TIL

But what if someonde does downgrade and reuse the nodes.conf, is this detected in any way or is it silently accepted? (This is a different discussion.)